src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs

```diff
@@ -493,12 +536,11 @@ public void UpdateConfigurations(DatabaseRecord databaseRecord)
 
             if (databaseRecord.PeriodicBackups == null)
             {
-                foreach (var periodicBackup in _periodicBackups)
+                Parallel.ForEach(_periodicBackups, (periodicBackup) =>
```

Comment:
In that case, if we had a DisposeOnce implementation with an async Dispose, we could write something much nicer